BuildRadiator/Client/Controllers/DashboardController.js

```diff
@@ -9,49 +9,93 @@
     ctrl.committerLimit = 11;
 
     function onMessageUpdate( message ) {
-      var tile = ctrl.tiles.find( function( tile ) {
+      var tiles = ctrl.tiles.filter( function( tile ) {
         return tile.type === 'message'
           && tile.config.messageKey === message.key;
       } );
 
-      if ( !tile ) {
-        return;
+      if ( tiles.length ) {
+        message.contentHtml = $sce.trustAsHtml( message.content );
       }
 
-      message.contentHtml = $sce.trustAsHtml( message.content );
-
-      tile.message = message;
+      tiles.forEach( function( t ) {
+        t.message = message;  
+      } );
     };
 
     function onProjectUpdateError( build ) {
-      var tile = ctrl.tiles.find( function( tile ) {
+      var projectTiles = ctrl.tiles.filter( function( tile ) {
         return tile.type === 'project'
           && tile.config.buildId === build.buildId
           && tile.config.branchName === build.branchName;
       } );
 
-      if ( !tile ) {
-        $log.error( 'UNKNOWN PROJECT', build );
-        return;
-      }
+      projectTiles.forEach( function( t ) {
+        t.error = build.error;
+      } );
+      
+      var primaryProjectTiles = ctrl.tiles.filter( function( tile ) {
+        return tile.type === 'dual-project'
+          && tile.config.primaryBuildId === build.buildId
+          && tile.config.primaryBranchName === build.branchName;
+      } );
 
-      tile.error = build.error;
+      primaryProjectTiles.forEach( function( t ) {
+        t.primaryError = build.error;
+      } );
+      
+      var secondaryProjectTiles = ctrl.tiles.filter( function( tile ) {
+        return tile.type === 'dual-project'
+          && tile.config.secondaryBuildId === build.BuildId
+          && tile.config.secondaryBranchName === build.branchName;
+      } );
+
+      secondaryProjectTiles.forEach( function( t ) {
+        t.secondaryError = build.error;
+      } );
+
+      if ( !projectTiles.length && !primaryProjectTiles.length && !secondaryProjectTiles.length ) {
+        $log.error( 'onProjectError: UNKNOWN PROJECT', build );
+      }
     };
 
     function onProjectUpdate( build ) {
-      var tile = ctrl.tiles.find( function( tile ) {
+      var projectTiles = ctrl.tiles.filter( function( tile ) {
```

Comment:
Updated to use `filter` in case the same build was placed multiple times on the same page. Not used in our set up but could be useful for the future.